src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdkAppEngineDeployment.java

```diff
@@ -48,12 +53,22 @@ public void deploy(DeployConfiguration config) throws AppEngineException {
 
     List<String> arguments = new ArrayList<>();
     arguments.add("deploy");
-    for (File deployable : config.getDeployables()) {
-      if (!deployable.exists()) {
-        throw new IllegalArgumentException(
-            "Deployable " + deployable.toPath().toString() + " does not exist.");
+    // If we get a single directory, then look in it for yamls
+    if (config.getDeployables().size() == 1 && config.getDeployables().get(0).isDirectory()) {
+      for (String filename : yamlFilenames) {
+        File yamlFile = new File(config.getDeployables().get(0), filename);
```

Comment:
I would put `config.getDeployables().get(0)` into a local variable so you don't have to repeat it.